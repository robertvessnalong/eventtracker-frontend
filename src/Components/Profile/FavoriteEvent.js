import { useContext, useState, useEffect } from 'react';
import UserContext from '../../Context/UserContext';
import { Row, Col, Spinner, Card, Button } from 'react-bootstrap';
import uuid from 'react-uuid';

const FavoriteEvent = () => {
  const { user } = useContext(UserContext);
  const [favorites, setFavorites] = useState([]);
  const [loading, isLoading] = useState(true);

  useEffect(() => {
    const setUserFavorites = () => {
      isLoading(false);
      const events = user.favorites.filter(({ type }) => type === 'event');
      setFavorites(events);
    };
    if (user.favorites) {
      setUserFavorites();
    }
  }, [user.favorites]);

  if (loading) {
    return (
      <Row>
        <Col>
          <Spinner animation='border' role='status'>
            <span className='visually-hidden'>Loading...</span>
          </Spinner>
        </Col>
      </Row>
    );
  }

  if (favorites.length < 1) {
    return (
      <Row>
        <Col>
          <span>You do not have any favorties yet!</span>
        </Col>
      </Row>
    );
  }

  return (
    <Row>
      <Col>
        {favorites.map((favorite) => (
          <Card key={uuid()} className='Favorite mb-4'>
            <Card.Header>{favorite.eventName}</Card.Header>
            <Card.Body>
              <Card.Text as='div'>
                <div className='card-time'>
                  <b>{favorite.eventDate}</b>
                </div>
                {favorite.eventAddress}
                <br />
                {favorite.eventExtendedAddress}
              </Card.Text>
              <a href={favorite.url} rel='noreferrer' target='_blank'>
                <Button className='blue-btn' variant='primary'>
                  Buy Tickets
                </Button>
              </a>
            </Card.Body>
          </Card>
        ))}
      </Col>
    </Row>
  );
};

export default FavoriteEvent;
