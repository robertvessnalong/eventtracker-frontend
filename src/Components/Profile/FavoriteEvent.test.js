import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FavoriteEvent from './FavoriteEvent';
import TestUser from '../../Testing/TestUser';
import UserContext from '../../Context/UserContext';
import { BrowserRouter } from 'react-router-dom';

describe('Favorite Performer Component', () => {
  // Smoke Test
  test('renders without crashing', async () => {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <FavoriteEvent />
        </UserContext.Provider>
      </BrowserRouter>
    );

    const label = screen.getByText(/BKFC Fight Night Wichita Rickels Vs Lane/i);
    expect(label).toBeInTheDocument();
  });

  // Snapshot
  test('matches snapshot', async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <FavoriteEvent />
        </UserContext.Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
