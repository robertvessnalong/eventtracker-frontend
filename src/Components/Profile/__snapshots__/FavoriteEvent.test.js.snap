// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Performer Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col"
    >
      <div
        class="Favorite mb-4 card"
      >
        <div
          class="card-header"
        >
          BKFC Fight Night Wichita Rickels Vs Lane
        </div>
        <div
          class="card-body"
        >
          <div
            class="card-text"
          >
            <div
              class="card-time"
            >
              <b>
                Sun Oct 24 2021 - 12 AM
              </b>
            </div>
            8151 N. Hartman Arena Drive
            <br />
            Park City, KS 67147
          </div>
          <a
            rel="noreferrer"
            target="_blank"
          >
            <button
              class="blue-btn btn btn-primary"
              type="button"
            >
              Buy Tickets
            </button>
          </a>
        </div>
      </div>
      <div
        class="Favorite mb-4 card"
      >
        <div
          class="card-header"
        >
          Struggle Jennings
        </div>
        <div
          class="card-body"
        >
          <div
            class="card-text"
          >
            <div
              class="card-time"
            >
              <b>
                Sat Oct 30 2021 - 12 AM
              </b>
            </div>
            332 East 1st Street North
            <br />
            Wichita, KS 67202
          </div>
          <a
            rel="noreferrer"
            target="_blank"
          >
            <button
              class="blue-btn btn btn-primary"
              type="button"
            >
              Buy Tickets
            </button>
          </a>
        </div>
      </div>
      <div
        class="Favorite mb-4 card"
      >
        <div
          class="card-header"
        >
          Mercyme with Jeremy Camp
        </div>
        <div
          class="card-body"
        >
          <div
            class="card-text"
          >
            <div
              class="card-time"
            >
              <b>
                Sat Oct 30 2021 - 12 AM
              </b>
            </div>
            8151 N. Hartman Arena Drive
            <br />
            Park City, KS 67147
          </div>
          <a
            rel="noreferrer"
            target="_blank"
          >
            <button
              class="blue-btn btn btn-primary"
              type="button"
            >
              Buy Tickets
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
